astropy/io/fits/column.py

```diff
@@ -320,6 +320,10 @@ def __new__(cls, format, strict=False):
         self.format, self.width, self.precision = \
             _parse_ascii_tformat(format, strict)
 
+        # If no width has been specified, set the dtype here to default as well
+        if format == self.format:
+            self.recformat = ASCII2NUMPY[format]
```

Comment:
Why this change ? A default width should already been set by `_parse_ascii_tformat` if needed, and the `recformat` property will use this as well.

Reply:
See my comment about the [race condition](https://github.com/astropy/astropy/pull/9901#issuecomment-580208749) -`_parse_ascii_tformat` will first set the default `width` to `10`, but then infer a required `recformat` of `i8` for that. But `I10` is not even sufficient to fully represent `i4`, which seems problematic when creating a new column to be written as ascii table. `test_column_format_interpretation` created the default `fits.Column('TEST', 'I', ascii=True)` with `i4`, although that was not actually tested. But I decided to leave it at that for now.
OTOH when creating a new column to be written to a file, we might not care about the exact `recformat`, since it comes out `ASCII`-formatted in the end anyway. From that point it might be preferable to set `i8` and `I11` as the defaults, which would support the full range of signed `int32` for writing and provide extra internal precision for any operations one might perform on the column data before writing. But writing as `I11` per default would of course mean changing output.

Reply:
I spend some time to remember myself this code, and sorry I don't understand what you mean here.
I don't see a case where your change would fix something that is done by `_parse_ascii_tformat` ?

Reply:
That said I think most issues are fixed by changing the default `recformat` to `i8` in `ASCII2NUMPY`. And I think we should also change the default to `I11` as well.